src/ip_data.py

```diff
@@ -78,10 +80,94 @@ def interleave(iters: list[Iterator[T]]) -> Iterator[T]:
 
 
 def repeat(genrec: Callable[[], T], times: int) -> Iterator[T]:
-    for i in range(times):
+    for _ in range(times):
         yield genrec()
 
 
+class FileWriter(ABC):
+    def __init__(self, base_filename: str):
+        self.base_filename = base_filename
+
+    @abstractmethod
+    def write_access(self, generator: Iterator[List[Any]]) -> None:
+        pass
+
+    @abstractmethod
+    def write_people(self, generator: Iterator[List[Any]]) -> None:
+        pass
+
+    @abstractmethod
+    def write_subnet(self, generator: Iterator[List[Any]]) -> None:
+        pass
+
+
+class JSONFileWriter(FileWriter):
+    def _write_to_json(self, filename: str, schema: List[str], generator: Iterator[List[Any]]) -> None:
+        result = []
+        for record in generator:
+            result.append(dict(zip(schema, map(str, record))))
+        with open(f"{self.base_filename}_{filename}.json", mode="w") as file:
+            json.dump(result, file, indent=2)
+
+    def write_access(self, generator: Iterator[List[Any]]) -> None:
+        self._write_to_json("access", ACCESS_SCHEMA, generator)
+
+    def write_people(self, generator: Iterator[List[Any]]) -> None:
+        self._write_to_json("people", PEOPLE_SCHEMA, generator)
+
+    def write_subnet(self, generator: Iterator[List[Any]]) -> None:
+        self._write_to_json("subnet", SUBNET_SCHEMA, generator)
+
+
+class SqliteWriter(FileWriter):
+    def __init__(self, base_filename: str):
+        super().__init__(base_filename)
+        self.db = sqlite3.connect(f"{self.base_filename}.db")
+        self.cursor = self.db.cursor()
+
+    def __enter__(self) -> Self:
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:
+        self.db.commit()
+        self.db.close()
```

Comment:
this won't work as you expect, define a `close` method on the `FileWriter` and call those in `MockDB.__exit__`

Reply:
you did not address this at all